api/jobs_test.go

```diff
@@ -260,6 +260,291 @@ var _ = Describe("Jobs API", func() {
 		})
 	})
 
+	Describe("GET /api/v1/pipelines/:pipeline_name/jobs/:job_name/badge", func() {
+		var response *http.Response
+
+		JustBeforeEach(func() {
+			var err error
+
+			response, err = client.Get(server.URL + "/api/v1/pipelines/some-pipeline/jobs/some-job/badge")
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(pipelineDBFactory.BuildWithTeamNameAndNameCallCount()).To(Equal(1))
+			teamName, pipelineName := pipelineDBFactory.BuildWithTeamNameAndNameArgsForCall(0)
+			Expect(pipelineName).To(Equal("some-pipeline"))
+			Expect(teamName).To(Equal(atc.DefaultTeamName))
+		})
+
+		Context("when getting the job config succeeds", func() {
+			BeforeEach(func() {
+				pipelineDB.GetConfigReturns(atc.Config{
+					Groups: []atc.GroupConfig{
+						{
+							Name: "group-1",
+							Jobs: []string{"some-job"},
+						},
+						{
+							Name: "group-2",
+							Jobs: []string{"some-job"},
+						},
+					},
+
+					Jobs: []atc.JobConfig{
+						{
+							Name: "some-job",
+							Plan: atc.PlanSequence{
+								{
+									Get: "some-input",
+								},
+								{
+									Get:      "some-name",
+									Resource: "some-other-input",
+									Params:   atc.Params{"secret": "params"},
+									Passed:   []string{"a", "b"},
+									Trigger:  true,
+								},
+								{
+									Put: "some-output",
+								},
+								{
+									Put:    "some-other-output",
+									Params: atc.Params{"secret": "params"},
+								},
+							},
+						},
+					},
+				}, 1, true, nil)
+			})
+
+			Context("when the finished build is successful", func() {
+				BeforeEach(func() {
+					pipelineDB.GetJobFinishedAndNextBuildReturns(
+						&db.Build{
+							ID:           1,
+							Name:         "1",
+							JobName:      "some-job",
+							PipelineName: "some-pipeline",
+							Status:       db.StatusSucceeded,
+							StartTime:    time.Unix(1, 0),
+							EndTime:      time.Unix(100, 0),
+						},
+						&db.Build{
+							ID:           3,
+							Name:         "2",
+							JobName:      "some-job",
+							PipelineName: "some-pipeline",
+							Status:       db.StatusStarted,
+						},
+						nil,
+					)
+				})
+
+				It("fetches by job", func() {
```

Comment:
This and the duplicate `It` should be moved up to line 318; GetJobFinishedAndNextBuild being called should be asserted on once; the following contexts ("when it's successful, when it is failed") then set up what happens afterwards.